hist/hist/src/TF1.cxx

```diff
@@ -3618,8 +3618,37 @@ void TF1::Streamer(TBuffer &b)
 
 void TF1::Update()
 {
-   delete fHistogram;
```

Comment:
This may be a very naive question, but why can't we directly make a full copy of `fHistogram` into the new one? I suppose at that point one could avoid deleting the previous histogram altogether. I acknowledge that if this function removes the histogram there must be a reason, but it is not immediately clear just by reading the body of the function

Reply:
You cannot just copy the histogram, because axis limits and bins can be different. Maybe one can just set new axes, but then again the axes attributes needs to be preserved